src/mjolnir/graphtilebuilder.cc

```diff
@@ -930,6 +930,7 @@ void GraphTileBuilder::AddBins(const std::string& tile_dir,
     auto bin = tile->GetBin(i % kBinsDim, i / kBinsDim);
     bins[i].assign(bin.begin(), bin.end());
     bins[i].insert(bins[i].end(), more_bins[i].cbegin(), more_bins[i].cend());
+    std::sort(bins[i].begin(), bins[i].end());
```

Comment:
the main fix

Reply:
this has a very unfortunate side effect which is that edges in the bin that are in this tile wont be the first edges in the bin. i would argue that this is not a "sensible" sorting of the data for that reason. we can work around it by providing our own sort lambda here though.

Reply:
the sort lambda could look like this:

```c++
[&tile_id](const GraphId& edge_a, const GraphId& edge_b) {
  // this tile always wins
  if(edge_a.tile_base() == tile_id && edge_b.tile_base() != tile_id)
    return true;
  // this tile always wins
  if(edge_b.tile_base() == tile_id && edge_a.tile_base != tile_id)
    return false;
  
  // either they are both in the same tile or they are both not in this tile
  // in either case we want to sort by level then by tile then by id but the
  // bits are arranged backwards for that so we flip them
  return edge_a.flip() < edge_b.flip();
}
```

and then we make a flip function in graphid.cc because this is useful:

```c++
  /**
   * Flips the bit order from id/tile/level to level/tile/id, so that sorting makes more sense
  *  The ordering of the bits in the id could be thought of as a design oversight
   * @return   Returns the graphid but with the fields in flipped order
   */
uint64_t flip() const {
  return static_cast<uint64_t>(id()) | (static_cast<uint64_t>(tileid()) << 21) | (static_cast<uint64_t>(level()) << 25);
}
```

we could then find where we do sorting like this (i think it must be all over the code) and use the flip function instead: https://github.com/valhalla/valhalla/blob/master/src/loki/node_search.cc#L209